### PR TITLE
NonPrintableCharacterFixer - fix for backslash and quotes when changing to escape sequences

### LINKED ISSUE #4663: "non_printable_character" is throwing linting exception
#### The PHP version you are using (`$ php -v`):
=> 7.3.11

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> 2.16.0

This file: https://github.com/PrestaShop/PrestaShop/blob/1.6.1.x/install-dev/langs/zh/install.php

can not be fixed with this very core/simple rule:
```
        'non_printable_character' => [
            'use_escape_sequences_in_strings' => true,
        ],
```

CLI error output:
```
        [PhpCsFixer\Linter\LintingException]                                                                            
        PHP Parse error: syntax error, unexpected '">点击这里</a>重新启动' (T_CONSTANT_ENCAPSED_STRING), expecting ')' on line 170.
```

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes #4663 and replaces #4664.

Ping @mvorisek as reporter and @ivan1986 as the original author for review.

### CODE REVIEW

**Comment by mvorisek** [comment]
For me it seems like it does not handle backslashes correctly:
```
$content = "'\\\\\\\\'";
var_dump($content);

$content = str_replace("\\'", "'", $content);
$content = preg_replace('/([\\\\$])/', '\\\\$1', $content);
var_dump($content);
```

**Comment by ivan1986** [comment]
Yes, need add tests
```
            [
                <<<'EXPECTED'
<?php echo "\\\u{200b}\"";
EXPECTED
                ,
                sprintf(<<<'INPUT'
<?php echo '\\%s"';
INPUT
                 , pack('H*', 'e2808b')),
            ],
            [
                <<<'EXPECTED'
<?php echo "\\\u{200b}'";
EXPECTED
                ,
                sprintf(<<<'INPUT'
<?php echo '\\%s\'';
INPUT
                 , pack('H*', 'e2808b')),
            ],

```
for check double slashes, it's fail now, or add slashes into exist tests

**Comment by kubawerlos** [comment]
> For me it seems

@mvorisek take an example from @ivan1986 - presenting test case is much more valuable.


@ivan1986 added yours 2 - thanks - and few more for backslashes.

**Comment by mvorisek** [comment]
> 
> 
> > For me it seems
> 
> @mvorisek take an example from @ivan1986 - presenting test case is much more valuable.
> 
> @ivan1986 added yours 2 - thanks - and few more for backslashes.

@kubawerlos Yes, but I hope I helped!

File: src/Fixer/Basic/NonPrintableCharacterFixer.php [hunk 1]
```diff
@@ -150,12 +151,20 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
                         $stringTypeChanged = true;
                     }
                 } elseif ("'" === $content[0]) {
-                    $content = Preg::replace('/^\'(.*)\'$/', '"$1"', $content);
                     $stringTypeChanged = true;
+                    $swapQuotes = true;
                 }
 
+                if ($swapQuotes) {
+                    $content = str_replace("\\'", "'", $content);
```
Comment (mvorisek) [inline src/Fixer/Basic/NonPrintableCharacterFixer.php]: Will this handle correctly `'\\'`?

**Comment by julienfalque** [comment]
Thank you @kubawerlos.

**Pull request merged**